lib/core/config.rb

```diff
@@ -2,22 +2,27 @@
 
 class Config # rubocop:disable Metrics/ClassLength
   attr_reader :config, :current,
-              :org, :org_comes_from_env, :app, :apps, :app_dir,
+              :org, :org_comes_from_env, :app, :apps, :app_dir, :required_options,
               # command line options
               :args, :options
 
   CONFIG_FILE_LOCATIION = ".controlplane/controlplane.yml"
 
-  def initialize(args, options)
+  def initialize(args, options, required_options)
```

Comment:
Isn't it better we handle this in commands?
Commands should care about the options they must receive.

Reply:
We're already handling `args` and `options` here, so it makes sense to handle `required_options` here too.

Reply:
This way, we make it more difficult to change in the future.
I think we should handle this issue within each command.
The command should be responsible for checking if it gets what it is dependent on.

I am just sharing my thoughts. You have more knowledge of OOP.

Reply:
Handling this within each command will add a lot of verbose and repeated code.

I don't think the command should be responsible for checking if it gets what it depends on, it should only do what it needs to do and assume that all args/options will be there and will be type-validated (that's the whole point of using Thor). Anything more specific than that, sure, the command should handle it.

If we were to move this to each command we'd essentially have this in every single command:

```
unless options[:app] do
  raise "..."
end
```

Why not do it in a single place? It actually makes it easier to change it in the future.